go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -1670,28 +1670,115 @@ func SystemVariablesInConfig(conf config.ReadableConfig) ([]sql.SystemVariable,
 
 var initMu = sync.Mutex{}
 
+// InitPersistedSystemVars loads all persisted global variables from disk and initializes the corresponding
+// SQL system variables with their values.
 func InitPersistedSystemVars(dEnv *env.DoltEnv) error {
 	initMu.Lock()
 	defer initMu.Unlock()
 
-	var globals config.ReadWriteConfig
+	// Find all the persisted global vars and load their values into sql.SystemVariables
+	persistedGlobalVars, err := findPersistedGlobalVars(dEnv)
+	if err != nil {
+		return err
+	}
+	sql.SystemVariables.AddSystemVariables(persistedGlobalVars)
+	return nil
+}
+
+// PersistSystemVarDefaults persists any SQL system variables that have non-deterministic default values, and
+// must have their generated default value persisted to disk. If the system variable is already persisted to disk,
+// then no changes are made. Currently, the only SQL system variable that requires persisting its default value
+// is @@server_uuid, since we need a consistent value used each time the server is started.
+func PersistSystemVarDefaults(dEnv *env.DoltEnv) error {
+	initMu.Lock()
+	defer initMu.Unlock()
+
+	// Find all the persisted global vars and load their values into sql.SystemVariables
+	persistedGlobalVars, err := findPersistedGlobalVars(dEnv)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the @@server_uuid value is persisted
+	var globalConfig config.ReadWriteConfig
 	if localConf, ok := dEnv.Config.GetConfig(env.LocalConfig); ok {
-		globals = config.NewPrefixConfig(localConf, env.SqlServerGlobalsPrefix)
+		globalConfig = config.NewPrefixConfig(localConf, env.SqlServerGlobalsPrefix)
 	} else if globalConf, ok := dEnv.Config.GetConfig(env.GlobalConfig); ok {
-		globals = config.NewPrefixConfig(globalConf, env.SqlServerGlobalsPrefix)
+		globalConfig = config.NewPrefixConfig(globalConf, env.SqlServerGlobalsPrefix)
 	} else {
+		return fmt.Errorf("unable to find local or global Dolt configuration")
+	}
+	return persistServerUuid(persistedGlobalVars, globalConfig)
+}
+
+// findPersistedGlobalVars searches the local and global configuration for the specified Dolt environment |dEnv| and
+// finds all global persisted system variables. Since global system vars can be persisted in either the local or
+// global configuration stores, this function searches both.
+func findPersistedGlobalVars(dEnv *env.DoltEnv) (persistedGlobalVars []sql.SystemVariable, err error) {
```

Comment:
It seems very odd that any of these functions operate on a single doltEnv, which necessarily limits them to a single DB. The server_id is a server-level concern, and is therefore never appropriate to store in any particular env.

Don't we have a file in `.doltcfg` where we keep these things now? I've been assuming that the config-based storage for persisted vars is basically defunct at this point.

Reply:
As far as I could tell... global system variables were previously always stored in `~/.dolt/config-global.json`, which was crummy because it made it global for a machine, not global for a sql-server. Someone changed that a while back so that if there is a local config dir set up, then `DATADIR/.dolt/config-global.json` is used, which solves the problem of persisting something globally for a sql-server, but not globally across all sql-servers on the host. 

The existing pattern we have from `InitPersistedSystemVars` also loads from a single DoltEnv rooted at `DATADIR`, so this is just extending that logic by using the DoltEnv passed into that function. 

Happy to talk about better ways to persist globals sys vars though. I can see using `.doltcfg` being a better solution than `.dolt`, but that's what we've got today.